content/articles/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/index.md

```diff
@@ -0,0 +1,146 @@
+interface = devices.Activate(IAudioEndpointVolume._iid_, CLSCTX_ALL, None)
+volume = cast(interface, POINTER(IAudioEndpointVolume))
+
+volMin, volMax = volume.GetVolumeRange()[:2]
+
+while True:
+    success, img = cap.read()  # if the camera we have specified works, we will capture an image.
+    imgRGB = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)  # converting the image the camera captures to RGB Image.
+    results = hands.process(imgRGB)  # completes the processing of the converted image.
+
+    lmList = []  # creating an empty list.
+    if results.multi_hand_landmarks:  # A list of all the detected hands.
+        #  Creating a loop so we can the list above so as to get the corresponding flag bit of each hand.
+        for handlandmark in results.multi_hand_landmarks:
+            for id, lm in enumerate(handlandmark.landmark):  # We are adding a counter and then returning it.
+                # Finding the finger joints of the hand
+                h, w, _ = img.shape
+                cx, cy = int(lm.x * w), int(lm.y * h)
+                lmList.append([id, cx, cy])  # Adding to the empty list we created "lmlist".
+            mpDraw.draw_landmarks(img, handlandmark, mpHands.HAND_CONNECTIONS)  # We draw all the landmarks in the frame
+
+    if lmList != []:
+        x1, y1 = lmList[4][1], lmList[4][2]  # specifying point 4 of the thumb. Refer to the hand image we discussed.
+        x2, y2 = lmList[8][1], lmList[8][2]  # specifying point 8 of the index finger. Refer to the hand image we
+                                             # discussed.
+        # We now create a circle around the thumb and the index finger.
+        cv2.circle(img, (x1, y1), 4, (255, 0, 0), cv2.FILLED)  # circle around the thumb.
+        cv2.circle(img, (x2, y2), 4, (255, 0, 0), cv2.FILLED)  # circle around the index finger.
+        cv2.line(img, (x1, y1), (x2, y2), (255, 0, 0), 3)  # Drawing a line between the index finger and the thumb.
+
+        length = hypot(x2 - x1, y2 - y1)  # Finding distance between the tips of the thumb and index finger using
+                                            # hypotenuse.
+        vol = np.interp(length, [15, 220], [volMin, volMax])  # Hand range 15 - 220
+        print(vol, length)
+        volume.SetMasterVolumeLevel(vol, None)  # Setting the master volume in accordance to hand range.
+                                                # Volume range -63.5 - 0.0
+
+    cv2.imshow('Image', img)  # Showing the real time video which shows how the user is adjusting the volume using their
+                                # hand gesture
+    if cv2.waitKey(1) & 0xff == ord('q'):  # Terminate the program if the user does press letter q.
+        break
+```
+### Results
+When the code above has run to completion without any errors, the results will be as those shown in the screenshot below.
+
+![Results](/engineering-education/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/results.png)
+

```

Comment:
You could provide a gif image demonstrating these results instead. Providing only a screenshot doesn't suffice.